### PR TITLE
libdokan: Handle deletions after renames properly

### PR DESCRIPTION
**Pull request by taruti**
Fix bug.

### CODE REVIEW

File: libdokan/dir.go [hunk 1]
```diff
@@ -486,6 +486,8 @@ func (d *Dir) Cleanup(ctx context.Context, fi *dokan.FileInfo) {
 	defer func() { d.folder.reportErr(ctx, libkbfs.WriteMode, err) }()
 
 	if fi != nil && fi.IsDeleteOnClose() && d.parent != nil {
+		d.folder.fs.renameAndDeletionLock.Lock()
```
Comment (zanderz) [inline libdokan/dir.go]: Unlocking with defer means that this lock is still held curing the call to forgetNode(), below, which I see attempts to acquire `f.mu`, which looks dangerous to me.

**Comment by taruti** [comment]
Added documentation about the order of locking.

**Comment by taruti** [comment]
Jenkins failure not related.

**Comment by zanderz** [comment]
Thank you! LGTM :thumbsup:

**Pull request merged**